src/frontend/src/components/Header/DesktopHeader.jsx

```diff
@@ -114,46 +115,48 @@ const Header = () => {
   );
 
   return (
-    <div>
-      <AppBar position="fixed" className={classes.root}>
-        <Toolbar>
-          <Typography variant="h3" className={classes.title}>
-            <Link to="/" title="Home" className={classes.title}>
-              {title}
-            </Link>
-          </Typography>
-          <IconButton color="inherit" className={classes.button}>
-            <Link to="/search">
-              <SearchIcon className={classes.searchIcon} />
-            </Link>
-          </IconButton>
-          <Button color="inherit" size="medium" className={classes.button}>
-            <Link to="/" className={classes.links}>
-              Home
-            </Link>
-          </Button>
-          <Login />
-          <IconButton
-            onClick={toggleDrawer('right', true)}
-            edge="start"
-            color="inherit"
-            aria-label="menu"
-            className={classes.button}
-          >
-            <MenuIcon className={classes.menuIcon} />
-          </IconButton>
-          <Drawer
-            classes={{ paper: classes.paper }}
-            anchor="right"
-            open={state.right}
-            onClose={toggleDrawer('right', false)}
-          >
-            {sideList('right')}
-          </Drawer>
-        </Toolbar>
-      </AppBar>
+    <>
+      <HideOnScroll>
```

Comment:
I think you should be wrapping this around https://github.com/Seneca-CDOT/telescope/blob/master/src/frontend/src/components/Header/Header.jsx#L14 vs. here and in the mobile one, so you don't have to duplicate.

Reply:
Theoretically, this is a very small and simple thing to do, it should just look like the following:
In Header.jsx
`import HideOnScroll from '../HideOnScroll';`
`...`
```
  return (
    <div>
      <HideOnScroll>
        {matches ? (
          <MobileHeader className="mobile-header" />
        ) : (
          <DesktopHeader className="header" />
        )}
      </HideOnScroll>
    </div>
  );
```

but no, I got a bunch of errors. I've been trying to fix it for like 2 hours. I got like 15 errors so I'm not gonna paste them here. I've tried React.forwardRef but have no idea if I was using it properly

Reply:
What are the errors?  You can paste huge text blocks in GitHub inside a details/summary block, see https://gist.github.com/ericclemmons/b146fe5da72ca1f706b2ef72a20ac39d.

Reply:
Everything builds but when I scroll down to where the Navbar is suppose to hide, the browser goes white and I get these errors

<details>
<summary>Errors</summary>

```md
:8000/commons.js:247020 404 page could not be found. Checkout https://www.gatsbyjs.org/docs/add-404-page/
__stack_frame_overlay_proxy_console__ @ :8000/commons.js:247020
:8000/commons.js:263887 [HMR] connected
:8000/commons.js:254615 React-Hot-Loader: react-🔥-dom patch is not detected. React 16.6+ features may not work.
AppContainer @ :8000/commons.js:254615
dev.telescope.cdot.systems/user/info:1 Failed to load resource: the server responded with a status of 403 ()
:8000/commons.js:247020 Error getting user info Error
    at _callee$ (:8000/commons.js:268193)
    at tryCatch (:8000/commons.js:259967)
    at Generator.invoke [as _invoke] (:8000/commons.js:260196)
    at Generator.prototype.<computed> [as next] (:8000/commons.js:260019)
    at asyncGeneratorStep (:8000/commons.js:5805)
    at _next (:8000/commons.js:5827)
__stack_frame_overlay_proxy_console__ @ :8000/commons.js:247020
:8000/commons.js:247020 Warning: Failed prop type: Invalid prop `children` supplied to `ForwardRef(Slide)`. Expected an element that can hold a ref. Did you accidentally use a plain function component for an element instead? For more information see https://material-ui.com/r/caveat-with-refs-guide
    in ForwardRef(Slide) (at HideOnScroll.jsx:14)
    in HideOnScroll (at Header.jsx:15)
    in div (at Header.jsx:14)
    in Header (at PageBase.js:32)
    in PageBase (created by WithStyles(PageBase))
    in WithStyles(PageBase) (at pages/index.js:156)
    in IndexPage (created by HotExportedIndexPage)
    in AppContainer (created by HotExportedIndexPage)
    in HotExportedIndexPage (created by PageRenderer)
    in PageRenderer (at query-result-store.js:86)
    in PageQueryStore (at root.js:56)
    in RouteHandler (at root.js:78)
    in div (created by FocusHandlerImpl)
    in FocusHandlerImpl (created by Context.Consumer)
    in FocusHandler (created by RouterImpl)
    in RouterImpl (created by Context.Consumer)
    in Location (created by Context.Consumer)
    in Router (created by EnsureResources)
    in ScrollContext (at root.js:69)
    in RouteUpdates (at root.js:68)
    in EnsureResources (at root.js:66)
    in LocationHandler (at root.js:124)
    in LocationProvider (created by Context.Consumer)
    in Location (at root.js:123)
    in Root (at root.js:131)
    in ThemeProvider (at TopLayout.js:18)
    in TopLayout (at gatsby-browser.js:6)
    in StaticQueryStore (at root.js:138)
    in _default (at app.js:67)
__stack_frame_overlay_proxy_console__ @ :8000/commons.js:247020
:8000/commons.js:247020 Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

Check the render method of `Transition`.
    in MobileHeader (at Header.jsx:17)
    in Transition (created by ForwardRef(Slide))
    in ForwardRef(Slide) (at HideOnScroll.jsx:14)
    in HideOnScroll (at Header.jsx:15)
    in div (at Header.jsx:14)
    in Header (at PageBase.js:32)
    in PageBase (created by WithStyles(PageBase))
    in WithStyles(PageBase) (at pages/index.js:156)
    in IndexPage (created by HotExportedIndexPage)
    in AppContainer (created by HotExportedIndexPage)
    in HotExportedIndexPage (created by PageRenderer)
    in PageRenderer (at query-result-store.js:86)
    in PageQueryStore (at root.js:56)
    in RouteHandler (at root.js:78)
    in div (created by FocusHandlerImpl)
    in FocusHandlerImpl (created by Context.Consumer)
    in FocusHandler (created by RouterImpl)
    in RouterImpl (created by Context.Consumer)
    in Location (created by Context.Consumer)
    in Router (created by EnsureResources)
    in ScrollContext (at root.js:69)
    in RouteUpdates (at root.js:68)
    in EnsureResources (at root.js:66)
    in LocationHandler (at root.js:124)
    in LocationProvider (created by Context.Consumer)
    in Location (at root.js:123)
    in Root (at root.js:131)
    in ThemeProvider (at TopLayout.js:18)
    in TopLayout (at gatsby-browser.js:6)
    in StaticQueryStore (at root.js:138)
    in _default (at app.js:67)
__stack_frame_overlay_proxy_console__ @ :8000/commons.js:247020
:8000/commons.js:27939 Uncaught TypeError: Cannot read property 'style' of null
    at Object.handleExit [as onExit] (:8000/commons.js:27939)
    at Transition.performExit (:8000/commons.js:257205)
    at Transition.performExit (:8000/commons.js:252954)
    at Transition.updateStatus (:8000/commons.js:257147)
    at Transition.updateStatus (:8000/commons.js:252954)
    at Transition.componentDidUpdate (:8000/commons.js:257108)
    at Transition.componentDidUpdate (:8000/commons.js:252944)
    at commitLifeCycles (:8000/commons.js:238350)
    at commitLayoutEffects (:8000/commons.js:241318)
    at HTMLUnknownElement.callCallback (:8000/commons.js:218703)
    at Object.invokeGuardedCallbackDev (:8000/commons.js:218752)
    at invokeGuardedCallback (:8000/commons.js:218807)
    at commitRootImpl (:8000/commons.js:241056)
    at unstable_runWithPriority (:8000/commons.js:244598)
    at runWithPriority$1 (:8000/commons.js:229554)
    at commitRoot (:8000/commons.js:240896)
    at finishSyncRender (:8000/commons.js:240322)
    at performSyncWorkOnRoot (:8000/commons.js:240308)
    at :8000/commons.js:229604
    at unstable_runWithPriority (:8000/commons.js:244598)
    at runWithPriority$1 (:8000/commons.js:229554)
    at flushSyncCallbackQueueImpl (:8000/commons.js:229599)
    at flushSyncCallbackQueue (:8000/commons.js:229587)
    at scheduleUpdateOnFiber (:8000/commons.js:239714)
    at dispatchAction (:8000/commons.js:234175)
    at handleScroll (:8000/commons.js:40990)
:8000/commons.js:247020 The above error occurred in the <Transition> component:
    in Transition (created by ForwardRef(Slide))
    in ForwardRef(Slide) (at HideOnScroll.jsx:14)
    in HideOnScroll (at Header.jsx:15)
    in div (at Header.jsx:14)
    in Header (at PageBase.js:32)
    in PageBase (created by WithStyles(PageBase))
    in WithStyles(PageBase) (at pages/index.js:156)
    in IndexPage (created by HotExportedIndexPage)
    in AppContainer (created by HotExportedIndexPage)
    in HotExportedIndexPage (created by PageRenderer)
    in PageRenderer (at query-result-store.js:86)
    in PageQueryStore (at root.js:56)
    in RouteHandler (at root.js:78)
    in div (created by FocusHandlerImpl)
    in FocusHandlerImpl (created by Context.Consumer)
    in FocusHandler (created by RouterImpl)
    in RouterImpl (created by Context.Consumer)
    in Location (created by Context.Consumer)
    in Router (created by EnsureResources)
    in ScrollContext (at root.js:69)
    in RouteUpdates (at root.js:68)
    in EnsureResources (at root.js:66)
    in LocationHandler (at root.js:124)
    in LocationProvider (created by Context.Consumer)
    in Location (at root.js:123)
    in Root (at root.js:131)
    in ThemeProvider (at TopLayout.js:18)
    in TopLayout (at gatsby-browser.js:6)
    in StaticQueryStore (at root.js:138)
    in _default (at app.js:67)

React will try to recreate this component tree from scratch using the error boundary you provided, AppContainer.
__stack_frame_overlay_proxy_console__ @ :8000/commons.js:247020
:8000/commons.js:247020 TypeError: Cannot read property 'style' of null
    at Object.handleExit [as onExit] (:8000/commons.js:27939)
    at Transition.performExit (:8000/commons.js:257205)
    at Transition.performExit (:8000/commons.js:252954)
    at Transition.updateStatus (:8000/commons.js:257147)
    at Transition.updateStatus (:8000/commons.js:252954)
    at Transition.componentDidUpdate (:8000/commons.js:257108)
    at Transition.componentDidUpdate (:8000/commons.js:252944)
    at commitLifeCycles (:8000/commons.js:238350)
    at commitLayoutEffects (:8000/commons.js:241318)
    at HTMLUnknownElement.callCallback (:8000/commons.js:218703)
    at Object.invokeGuardedCallbackDev (:8000/commons.js:218752)
    at invokeGuardedCallback (:8000/commons.js:218807)
    at commitRootImpl (:8000/commons.js:241056)
    at unstable_runWithPriority (:8000/commons.js:244598)
    at runWithPriority$1 (:8000/commons.js:229554)
    at commitRoot (:8000/commons.js:240896)
    at finishSyncRender (:8000/commons.js:240322)
    at performSyncWorkOnRoot (:8000/commons.js:240308)
    at :8000/commons.js:229604
    at unstable_runWithPriority (:8000/commons.js:244598)
    at runWithPriority$1 (:8000/commons.js:229554)
    at flushSyncCallbackQueueImpl (:8000/commons.js:229599)
    at flushSyncCallbackQueue (:8000/commons.js:229587)
    at scheduleUpdateOnFiber (:8000/commons.js:239714)
    at dispatchAction (:8000/commons.js:234175)
    at handleScroll (:8000/commons.js:40990)
__stack_frame_overlay_proxy_console__ @ :8000/commons.js:247020
:8000/commons.js:254646 Uncaught TypeError: Cannot read property 'style' of null
    at Object.handleExit [as onExit] (:8000/commons.js:27939)
    at Transition.performExit (:8000/commons.js:257205)
    at Transition.performExit (:8000/commons.js:252954)
    at Transition.updateStatus (:8000/commons.js:257147)
    at Transition.updateStatus (:8000/commons.js:252954)
    at Transition.componentDidUpdate (:8000/commons.js:257108)
    at Transition.componentDidUpdate (:8000/commons.js:252944)
    at commitLifeCycles (:8000/commons.js:238350)
    at commitLayoutEffects (:8000/commons.js:241318)
    at HTMLUnknownElement.callCallback (:8000/commons.js:218703)
    at Object.invokeGuardedCallbackDev (:8000/commons.js:218752)
    at invokeGuardedCallback (:8000/commons.js:218807)
    at commitRootImpl (:8000/commons.js:241056)
    at unstable_runWithPriority (:8000/commons.js:244598)
    at runWithPriority$1 (:8000/commons.js:229554)
    at commitRoot (:8000/commons.js:240896)
    at finishSyncRender (:8000/commons.js:240322)
    at performSyncWorkOnRoot (:8000/commons.js:240308)
    at :8000/commons.js:229604
    at unstable_runWithPriority (:8000/commons.js:244598)
    at runWithPriority$1 (:8000/commons.js:229554)
    at flushSyncCallbackQueueImpl (:8000/commons.js:229599)
    at flushSyncCallbackQueue (:8000/commons.js:229587)
    at scheduleUpdateOnFiber (:8000/commons.js:239714)
    at dispatchAction (:8000/commons.js:234175)
    at handleScroll (:8000/commons.js:40990)
:8000/commons.js:247020 The above error occurred in the <AppContainer> component:
    in AppContainer (created by HotExportedIndexPage)
    in HotExportedIndexPage (created by PageRenderer)
    in PageRenderer (at query-result-store.js:86)
    in PageQueryStore (at root.js:56)
    in RouteHandler (at root.js:78)
    in div (created by FocusHandlerImpl)
    in FocusHandlerImpl (created by Context.Consumer)
    in FocusHandler (created by RouterImpl)
    in RouterImpl (created by Context.Consumer)
    in Location (created by Context.Consumer)
    in Router (created by EnsureResources)
    in ScrollContext (at root.js:69)
    in RouteUpdates (at root.js:68)
    in EnsureResources (at root.js:66)
    in LocationHandler (at root.js:124)
    in LocationProvider (created by Context.Consumer)
    in Location (at root.js:123)
    in Root (at root.js:131)
    in ThemeProvider (at TopLayout.js:18)
    in TopLayout (at gatsby-browser.js:6)
    in StaticQueryStore (at root.js:138)
    in _default (at app.js:67)

React will try to recreate this component tree from scratch using the error boundary you provided, LocationProvider.
__stack_frame_overlay_proxy_console__ @ :8000/commons.js:247020
:8000/commons.js:184944 Uncaught TypeError: Cannot read property 'style' of null
    at Object.handleExit [as onExit] (:8000/commons.js:27939)
    at Transition.performExit (:8000/commons.js:257205)
    at Transition.performExit (:8000/commons.js:252954)
    at Transition.updateStatus (:8000/commons.js:257147)
    at Transition.updateStatus (:8000/commons.js:252954)
    at Transition.componentDidUpdate (:8000/commons.js:257108)
    at Transition.componentDidUpdate (:8000/commons.js:252944)
    at commitLifeCycles (:8000/commons.js:238350)
    at commitLayoutEffects (:8000/commons.js:241318)
    at HTMLUnknownElement.callCallback (:8000/commons.js:218703)
    at Object.invokeGuardedCallbackDev (:8000/commons.js:218752)
    at invokeGuardedCallback (:8000/commons.js:218807)
    at commitRootImpl (:8000/commons.js:241056)
    at unstable_runWithPriority (:8000/commons.js:244598)
    at runWithPriority$1 (:8000/commons.js:229554)
    at commitRoot (:8000/commons.js:240896)
    at finishSyncRender (:8000/commons.js:240322)
    at performSyncWorkOnRoot (:8000/commons.js:240308)
    at :8000/commons.js:229604
    at unstable_runWithPriority (:8000/commons.js:244598)
    at runWithPriority$1 (:8000/commons.js:229554)
    at flushSyncCallbackQueueImpl (:8000/commons.js:229599)
    at flushSyncCallbackQueue (:8000/commons.js:229587)
    at scheduleUpdateOnFiber (:8000/commons.js:239714)
    at dispatchAction (:8000/commons.js:234175)
    at handleScroll (:8000/commons.js:40990)
:8000/commons.js:247020 The above error occurred in the <LocationProvider> component:
    in LocationProvider (created by Context.Consumer)
    in Location (at root.js:123)
    in Root (at root.js:131)
    in ThemeProvider (at TopLayout.js:18)
    in TopLayout (at gatsby-browser.js:6)
    in StaticQueryStore (at root.js:138)
    in _default (at app.js:67)

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://fb.me/react-error-boundaries to learn more about error boundaries.
__stack_frame_overlay_proxy_console__ @ :8000/commons.js:247020
:8000/commons.js:229617 Uncaught TypeError: Cannot read property 'style' of null
    at Object.handleExit [as onExit] (:8000/commons.js:27939)
    at Transition.performExit (:8000/commons.js:257205)
    at Transition.performExit (:8000/commons.js:252954)
    at Transition.updateStatus (:8000/commons.js:257147)
    at Transition.updateStatus (:8000/commons.js:252954)
    at Transition.componentDidUpdate (:8000/commons.js:257108)
    at Transition.componentDidUpdate (:8000/commons.js:252944)
    at commitLifeCycles (:8000/commons.js:238350)
    at commitLayoutEffects (:8000/commons.js:241318)
    at HTMLUnknownElement.callCallback (:8000/commons.js:218703)
    at Object.invokeGuardedCallbackDev (:8000/commons.js:218752)
    at invokeGuardedCallback (:8000/commons.js:218807)
    at commitRootImpl (:8000/commons.js:241056)
    at unstable_runWithPriority (:8000/commons.js:244598)
    at runWithPriority$1 (:8000/commons.js:229554)
    at commitRoot (:8000/commons.js:240896)
    at finishSyncRender (:8000/commons.js:240322)
    at performSyncWorkOnRoot (:8000/commons.js:240308)
    at :8000/commons.js:229604
    at unstable_runWithPriority (:8000/commons.js:244598)
    at runWithPriority$1 (:8000/commons.js:229554)
    at flushSyncCallbackQueueImpl (:8000/commons.js:229599)
    at flushSyncCallbackQueue (:8000/commons.js:229587)
    at scheduleUpdateOnFiber (:8000/commons.js:239714)
    at dispatchAction (:8000/commons.js:234175)
    at handleScroll (:8000/commons.js:40990)
```
</details>

Reply:
I think I might have figured it out

Reply:
So `HideOnScroll.jsx` wanted:
```
  return (
    <Slide appear={false} direction="down" in={!trigger}>
      <div>{children}</div>
    </Slide>
  );
```
and didn't want
```
  return (
    <Slide appear={false} direction="down" in={!trigger}>
      {children}
    </Slide>
  );
```

The `<div>` around `{children}` is important because `<Slide>` adds a ref and props to the single child. If there are multiple children then `Slide` will be mad because `children.props` is not defined and it tries to reference `children.props.style` hence all the `style undefined` errors.